### PR TITLE
Rewrite dockerfile to make the image smaller

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did
It used to take a long time to download the image because it was so big. As title described, to make the image smaller built by the Dockerfile, I rewrite the Dockerfile using multi-stage builds. Now 

```
# docker images
REPOSITORY                                                 TAG                 IMAGE ID            CREATED             SIZE
dfclient                                                   latest              b1a01a3df1a9        12 minutes ago      28.5MB
dragonflyoss/dfclient                                      v0.3.0_dev          79cf3514b9dc        3 weeks ago         842MB
```
Finally, the program also runs normally and the mirror size is significantly reduced

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
just Dockerfile

### Ⅳ. Describe how to verify it
```
# docker build -t build .

# docker run -d -p 65001:65001 dfclient

# docker ps
CONTAINER ID        IMAGE                                                            COMMAND             CREATED             STATUS              PORTS                              NAMES
e69a9b472edc        b1a01a3df1a9                                                     "./dfdaemon"        14 minutes ago      Up 14 minutes       0.0.0.0:65001->65001/tcp           wizardly_meitner

# curl 127.0.0.1:65001/env
map[dfPattern:[] home:/root/ param:{DfPath:/dfclient/dfget DFRepo:/root/.small-dragonfly/dfdaemon/data/ RateLimit:20M CallSystem:com_ops_dragonfly URLFilter:Signature&Expires&OSSAccessKeyId Notbs:true HostIP:127.0.0.1 Registry:}]
```

### Ⅴ. Special notes for reviews

By the way , `dfdaemon` will use `dfget -v` to test whether `dfget` is as expected, you will get error with that, use `dfget version`  instead.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+
+WORKDIR /dfclient
+
+COPY --from=builder /dfclient /dfclient
 
 # dfdaemon will listen 65001 in dafault.
 EXPOSE 65001
 
-ENTRYPOINT ["dfdaemon"]
+ENTRYPOINT [ "./df-daemon", "--registry", "https://index.docker.io" ]
```
Comment (allencloud) [inline Dockerfile]: If you have set the workdir, please avoid to use `./`. @Starnop

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+COPY . /go/src/github.com/dragonflyoss/Dragonfly
 
-WORKDIR /go/src/github.com/alibaba/Dragonfly/build
+RUN apk --no-cache add bash make gcc libc-dev git
 
-RUN ./build.sh client && cd client && make install
+RUN CGO_ENABLED=1 GOOS=linux go build -a -installsuffix cgo -o /dfclient/dfdaemon cmd/dfdaemon/main.go
+RUN CGO_ENABLED=1 GOOS=linux go build -a -installsuffix cgo -o /dfclient/dfget cmd/dfget/main.go
+
+FROM alpine
```
[the remark this answers is not in the document]
Comment (Starnop) [reply Dockerfile]: Done.

File: Dockerfile (hunk 1 above)
Comment (Starnop) [reply Dockerfile]: Linux will search all paths in the ENV $PATH which the current directory is not in , so we need it and I use the absolute path here instead.

**Comment by allencloud** [comment]
> We should make it possible for user to set command line arguments of dfdaemon.

Actually now in the Dockerfile, it does allow users to do that. PTAL. @lowzj

**Comment by allencloud** [comment]
LGTM

**Pull request merged**